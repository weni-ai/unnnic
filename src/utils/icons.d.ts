declare module '@/utils/icons' {
  const icons: Record<string, any>;
  export default icons;
} 