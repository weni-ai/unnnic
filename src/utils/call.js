import Vue from 'vue';
import Alert from '../components/Alert/Alert.vue';
import Modal from '../components/Modal/Modal.vue';

export default {
  callAlert({ props }) {
    const AlertClass = Vue.extend(Alert);
    const instance = new AlertClass({
      propsData: { ...props, onClose: () => { instance.$el.remove(); } },
      created() { this.$on(['close'], () => { instance.$el.remove(); }); },
    });
    instance.$mount();
    document.body.appendChild(instance.$el);
  },

  callModal({ props }) {
    const ModalClass = Vue.extend(Modal);
    const instance = new ModalClass({
      propsData: { ...props, onClose: () => { instance.$el.remove(); } },
      created() { this.$on(['close'], () => { instance.$el.remove(); }); },
    });
    instance.$mount();
    document.body.appendChild(instance.$el);
  },
};
