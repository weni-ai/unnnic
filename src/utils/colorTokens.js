import colors from '@/assets/scss/colors.scss?inline';
export default colors