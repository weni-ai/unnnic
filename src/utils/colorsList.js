export default [
  'background-solo',
  'background-sky',
  'background-grass',
  'background-carpet',
  'background-snow',
  'neutral-black',
  'neutral-darkest',
  'neutral-dark',
  'neutral-cloudy',
  'neutral-cleanest',
  'neutral-clean',
  'neutral-soft',
  'neutral-lightest',
  'neutral-light',
  'neutral-snow',
  'feedback-red',
  'feedback-green',
  'feedback-yellow',
  'feedback-blue',
  'feedback-grey',
  'aux-blue',
  'aux-purple',
  'aux-orange',
  'aux-lemon',
  'aux-pink',
  'aux-baby-blue',
  'aux-baby-yellow',
  'aux-baby-red',
  'aux-baby-green',
  'aux-baby-pink',
  'brand-weni',
  'brand-weni-dark',
  'brand-weni-soft',
  'brand-sec-dark',
  'brand-sec-soft',
  'brand-sec',
];
