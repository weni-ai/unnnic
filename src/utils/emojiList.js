export default {
  Grinning_Face: 'u1F600',
  Beaming_Face_with_Smiling_Eyes: 'u1F601',
  Face_with_Tears_of_Joy: 'u1F602',
  Grinning_Face_with_Big_Eyes: 'u1F603',
  Grinning_Face_with_Smiling_Eyes: 'u1F604',
  Grinning_Face_with_Sweat: 'u1F605',
  Grinning_Squinting_Face: 'u1F606',
  Smiling_Face_with_Halo: 'u1F607',
  Smiling_Face_with_Horns: 'u1F608',
  Winking_Face: 'u1F609',
  Neutral_Face: 'u1F610',
  Expressionless_Face: 'u1F611',
  Unamused_Face: 'u1F612',
  Downcast_Face_with_Sweat: 'u1F613',
  Pensive_Face: 'u1F614',
  Confused_Face: 'u1F615',
  Confounded_Face: 'u1F616',
  Kissing_Face: 'u1F617',
  Face_Blowing_a_Kiss: 'u1F618',
  Kissing_Face_with_Smiling_Eyes: 'u1F619',
  Angry_Face: 'u1F620',
  Pouting_Face: 'u1F621',
  Crying_Face: 'u1F622',
  Persevering_Face: 'u1F623',
  Face_with_Steam_From_Nose: 'u1F624',
  Sad_but_Relieved_Face: 'u1F625',
  Frowning_Face_with_Open_Mouth: 'u1F626',
  Anguished_Face: 'u1F627',
  Fearful_Face: 'u1F628',
  Weary_Face: 'u1F629',
  Anxious_Face_with_Sweat: 'u1F630',
  Face_Screaming_in_Fear: 'u1F631',
  Astonished_Face: 'u1F632',
  Flushed_Face: 'u1F633',
  Sleeping_Face: 'u1F634',
  Hushed_Face_with_Closed_Eyes: 'u1F635',
  Face_Without_Mouth: 'u1F636',
  Face_with_Medical_Mask: 'u1F637',
  Grinning_Cat_with_Smiling_Eyes: 'u1F638',
  Cat_with_Tears_of_Joy: 'u1F639',
  Weary_Cat: 'u1F640',
  Slightly_Frowning_Face: 'u1F641',
  Slightly_Smiling_Face: 'u1F642',
  Upside_Down_Face: 'u1F643',
  Face_with_Rolling_Eyes: 'u1F644',
};
