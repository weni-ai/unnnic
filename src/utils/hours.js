const hoursTimesOptions = [
  {
    label: '00:00',
    value: '00:00',
  },
  {
    label: '01:00',
    value: '01:00',
  },
  {
    label: '02:00',
    value: '02:00',
  },
  {
    label: '03:00',
    value: '03:00',
  },
  {
    label: '04:00',
    value: '04:00',
  },
  {
    label: '05:00',
    value: '05:00',
  },
  {
    label: '06:00',
    value: '06:00',
  },
  {
    label: '07:00',
    value: '07:00',
  },
  {
    label: '08:00',
    value: '08:00',
  },
  {
    label: '09:00',
    value: '09:00',
  },
  {
    label: '10:00',
    value: '10:00',
  },
  {
    label: '11:00',
    value: '11:00',
  },
  {
    label: '12:00',
    value: '12:00',
  },
  {
    label: '13:00',
    value: '13:00',
  },
  {
    label: '14:00',
    value: '14:00',
  },
  {
    label: '15:00',
    value: '15:00',
  },
  {
    label: '16:00',
    value: '16:00',
  },
  {
    label: '17:00',
    value: '17:00',
  },
  {
    label: '18:00',
    value: '18:00',
  },
  {
    label: '19:00',
    value: '19:00',
  },
  {
    label: '20:00',
    value: '20:00',
  },
  {
    label: '21:00',
    value: '21:00',
  },
  {
    label: '22:00',
    value: '22:00',
  },
  {
    label: '23:00',
    value: '23:00',
  },
];

export { hoursTimesOptions };
