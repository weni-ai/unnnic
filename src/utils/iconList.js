export default [
  'app-window-edit-1',
  'app-window-edit-2',
  'headphones-customer-support-2',
  'building-2-1',
  'expand-8-1',
  'flash-1-4',
  'phone-charger-1',
  'checkbox-disable',
  'headphones-customer-support-1',
  'arrow-button-down-1',
  'vip-crown-queen-2',
  'read-email-at-1',
  'flash-1-3',
  'time-clock-circle-1',
  'search-1',
  'single-neutral-2',
  'messaging-we-chat-3',
  'upload-bottom-1',
  'checkbox-select',
  'radio-default',
  'view-1-1',
  'photography-equipment-bag-1',
  'messaging-we-chat-2',
  'button-refresh-arrow-1',
  'information-circle-4',
  'science-fiction-robot-2',
  'cog-2',
  'check-square-2',
  'indicator',
  'close-1',
  'translate-2',
  'radio-selected',
  'science-fiction-robot-1',
  'cog-1',
  'check-square-1',
  'translate-1',
  'phone-3',
  'button-refresh-arrows-1',
  'chat-translate-1',
  'charger-1',
  'phone-4',
  'expand-full-1',
  'graph-stats-1',
  'check-circle-1-1',
  'messages-bubble-4',
  'hierarchy-3-3',
  'radio-disable',
  'view-off-1',
  'hierarchy-3-2',
  'single-neutral-actions-1',
  'checkbox-less',
  'arrow-right-1-1',
  'task-list-clock-1',
  'arrow-button-up-1',
  'house-1-1',
  'download-bottom-1',
  'loading-circle-1',
  'layout-dashboard-1',
  'messages-bubble-3',
  'alert-circle-1-1',
  'messages-bubble-1',
  'switch-default',
  'developer-community-github-1-1',
  'lock-2-1',
  'pencil-write-1',
  'book-address-1-2',
  'delete-1',
  'arrow-left-1-1',
  'add-1',
  'checkbox-default',
  'layout-dashboard-2',
  'download-thick-bottom-1',
  'gauge-dashboard-1',
  'logout-1-1',
  'gauge-dashboard-2',
  'switch-selected',
  'social-media-google-1',
  'social-media-facebook-1',
  'social-instagram-1',
  'alarm-bell-2',
  'check-circle-1-1-1',
  'alarm-bell-3',
  'alert-circle-1',
  'book-address-2',
  'graph-stats-ascend-2',
  'synchronize-arrow-clock-5',
  'keyboard-arrow-down-1',
  'keyboard-arrow-left-1',
  'email-action-unread-1',
  'synchronize-arrow-clock-4',
  'house-2-2',
  'copy-paste-1',
  'typing-1',
  'drag',
  'lock-unlock-1-1',
  'rating-star-1',
  'send-email-3-1',
  'switch-default-disabled',
  'switch-selected-disabled',
  'paginate-filter-text-1',
  'bin-1-1',
  'floppy-disk-1',
  'messaging-whatsapp-1',
  'export-1',
  'book-library-1',
  'book-library-2',
];
