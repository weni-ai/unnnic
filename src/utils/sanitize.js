/**
 * Decodes HTML entities (`&lt;` → `<`, `&gt;` → `>`, etc.).
 * @param {string} input
 * @returns {string}
 */
export function escapeHtml(input) {
  if (typeof input !== 'string') return '';
  return input.replace(/<\/?[a-zA-Z]+\b[^>]*>/g, '');
}

/**
 * Sanitizes HTML by removing disallowed tags and unsafe attributes.
 * @param {string} input
 * @param {Array<string>} allowedTags - List of allowed HTML tags.
 * @param {number} maxLength - Maximum length of the sanitized string.
 * @returns {string}
 */
export function sanitizeHtml(input, allowedTags = [], maxLength = 1000) {
  if (typeof input !== 'string') return '';

  // Limits text length
  if (input.length > maxLength) {
    input = input.substring(0, maxLength);
  }

  // Decodes HTML entities (&lt;, &gt;, &amp;)
  input = escapeHtml(input);

  // Removes **only** real HTML tags
  input = input.replace(/<\/?([a-z][a-z0-9]*)\b[^>]*>/gi, (match, tagName) => {
    return allowedTags.includes(tagName.toLowerCase()) ? match : '';
  });

  return input;
}

/**
 * Fully sanitizes a string by applying multiple layers of security:
 * 1. `sanitizeHtml` removes disallowed tags and attributes.
 * 2. `escapeHtml` ensures no encoded characters remain and removes any remaining tags.
 *
 * @param {string} input
 * @param {Array<string>} allowedTags - List of allowed HTML tags.
 * @param {number} maxLength - Maximum length of the sanitized string.
 * @returns {string}
 */
export function fullySanitize(input, allowedTags = [], maxLength = 1000) {
  if (typeof input !== 'string') return '';

  let sanitizedInput = sanitizeHtml(input, allowedTags, maxLength);

  return escapeHtml(sanitizedInput);
}

/**
 * Validates and sanitizes input text.
 * @param {string} input - The input string to validate.
 * @param {Array<string>} allowedTags - List of allowed HTML tags.
 * @param {number} maxLength - Maximum length of the sanitized string.
 * @returns {{ isValid: boolean, sanitized: string, errors: string[] }} - Validation result.
 */
export function validateInput(input, allowedTags = [], maxLength = 1000) {
  const errors = [];

  if (typeof input !== 'string') {
    errors.push('Input must be a string.');
    return { isValid: false, sanitized: '', errors };
  }

  if (input.length > maxLength) {
    errors.push(`Input exceeds the maximum length of ${maxLength} characters.`);
    input = input.substring(0, maxLength);
  }

  const sanitized = fullySanitize(input, allowedTags, maxLength);

  if (sanitized !== input) {
    errors.push(
      'Some HTML tags or attributes were removed for security reasons.',
    );
  }

  return {
    isValid: errors.length === 0,
    sanitized,
    errors,
  };
}
