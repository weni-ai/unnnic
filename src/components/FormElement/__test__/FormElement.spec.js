import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import UnnnicFormElement from '../FormElement.vue';
import UnnnicIcon from '../../Icon.vue';
import { fullySanitize } from '@/utils/sanitize';

vi.mock('@/utils/sanitize', () => ({
    fullySanitize: vi.fn((input) => input),
}));

describe('UnnnicFormElement.vue', () => {
    it('renders correctly', () => {
        const wrapper = mount(UnnnicFormElement);
        expect(wrapper.exists()).toBe(true);
    });

    it('matches the snapshot', () => {
        const wrapper = mount(UnnnicFormElement, {
            props: {
                label: 'Snapshot Test Label',
                error: 'Snapshot Error',
                message: 'Snapshot Message',
                disabled: false,
            },
            slots: {
                rightMessage: '<span>Right Message</span>',
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders the label when provided', () => {
        const label = 'Test Label';
        const wrapper = mount(UnnnicFormElement, {
            props: { label },
        });

        expect(wrapper.text()).toContain(label);
        expect(fullySanitize).toHaveBeenCalledWith(label);
    });

    it('applies the disabled class when the disabled prop is true', () => {
        const wrapper = mount(UnnnicFormElement, {
            props: { disabled: true },
        });

        expect(wrapper.classes()).toContain('unnnic-form-element--disabled');
    });

    it('renders the error message when provided', () => {
        const errorMessage = 'Error occurred!';
        const wrapper = mount(UnnnicFormElement, {
            props: { error: errorMessage },
        });

        expect(wrapper.text()).toContain(errorMessage);
        expect(fullySanitize).toHaveBeenCalledWith(errorMessage);
    });

    it('renders the UnnnicIcon when there is an error message', () => {
        const wrapper = mount(UnnnicFormElement, {
            props: { error: 'Error text' },
        });

        expect(wrapper.findComponent(UnnnicIcon).exists()).toBe(true);
    });

    it('renders the rightMessage slot correctly', () => {
        const wrapper = mount(UnnnicFormElement, {
            slots: {
                rightMessage: '<span class="test-right-message">Right Message</span>',
            },
        });

        expect(wrapper.find('.test-right-message').exists()).toBe(true);
    });

    it('renders the message when provided', () => {
        const message = 'This is a message';
        const wrapper = mount(UnnnicFormElement, {
            props: { message },
        });

        expect(wrapper.text()).toContain(message);
        expect(fullySanitize).toHaveBeenCalledWith(message);
    });

    it('validates the size prop correctly', () => {
        const consoleWarnSpy = vi.spyOn(console, 'warn').mockImplementation(() => { });

        const validSizes = ['md', 'sm'];
        validSizes.forEach((size) => {
            const wrapper = mount(UnnnicFormElement, { props: { size } });
            expect(wrapper.props().size).toBe(size);
        });

        mount(UnnnicFormElement, { props: { size: 'invalid' } });

        expect(consoleWarnSpy).toHaveBeenCalled();

        consoleWarnSpy.mockRestore();
    });


    it('computes shouldShowErrorSection correctly', async () => {
        const wrapper = mount(UnnnicFormElement, {
            props: { error: 'Error message' },
        });

        expect(wrapper.vm.shouldShowErrorSection).toBe(true);

        await wrapper.setProps({ error: true });
        expect(wrapper.vm.shouldShowErrorSection).toBe(false);

        const wrapperWithSlot = mount(UnnnicFormElement, {
            props: { error: true },
            slots: {
                rightMessage: '<span>Right message</span>',
            },
        });

        expect(wrapperWithSlot.vm.shouldShowErrorSection).toBe(true);
    });

});
