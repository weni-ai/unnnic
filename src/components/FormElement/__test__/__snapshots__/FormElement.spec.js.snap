// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UnnnicFormElement.vue > matches the snapshot 1`] = `
"<section data-v-9f8d6c86="" class="unnnic-form-element">
  <p data-v-9f8d6c86="" class="unnnic-form-element__label">Snapshot Test Label</p>
  <p data-v-9f8d6c86="" class="unnnic-form-element__error"><span data-v-26446d8e="" data-v-9f8d6c86="" class="material-symbols-rounded unnnic-icon-scheme--aux-red-500 unnnic-icon-size--sm" data-testid="material-icon" translate="no">warning</span> Snapshot Error<span data-v-9f8d6c86="" class="unnnic-form-element__right-message"><span>Right Message</span></span></p>
  <p data-v-9f8d6c86="" class="unnnic-form-element__message">Snapshot Message
    <!--v-if-->
  </p>
</section>"
`;
